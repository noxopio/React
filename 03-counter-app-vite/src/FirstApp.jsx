import PropTypes from "prop-types"



// const newMessge={
// message:'Hola Mundo'
// }
// const name = 'Neo'
// const getResult = (a, b) => {

//   return a + b;
// }
export const FirstApp = ({ title, subTitle ,name}) => {

  // console.log(props);
  return (
    <>
      {/* <h1>{name}</h1> */}
      <h1 data-testid= 'test-title'>{ title }</h1>
      <h2> {name}</h2>
      <h2>{subTitle}</h2>
      <h2>{subTitle}</h2>
      {/*       <p>{getResult(2, 7)}</p> */}
      {/* <code>{JSON.stringify(newMessge)}</code>   */}
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired
}

FirstApp.defaultProps={
name:'Neo si hay name',
subTitle: 'No hay subtitulos',
title: '"Hola, Soy Goku',
}