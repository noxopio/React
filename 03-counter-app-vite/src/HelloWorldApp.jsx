


export  const HelloWorldApp = () => {
  return (
    <h1>
      hello world
    </h1>
  )
}

