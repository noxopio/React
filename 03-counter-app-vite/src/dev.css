* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

section {
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 100vh;
  background-color: #1c1c1c;
}

.box {
  position: relative;
  width: 360px;
  height: 200px;
  /* background-color: #fff; */
  display: flex;
  justify-content: center;
  align-items: center;

}

.circle {
  position: relative;
  width: 200px;
  height: 200px;
  margin: 0 -20px;
  border: 40px solid #000;
  border-radius: 50%;
}

.dot {
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translateY(-50%);
  width: 40px;
  height: 40px;
  background-color: #0f0;
  border-radius: 50%;
  transform-origin: 100px 50%;
  animation: animatedot 4s linear infinite;
  /* color: rgb(238, 19, 19);  */
  animation-delay: calc(0.025s * var(--i));
  box-shadow: 0 0 30px rgba(248, 0, 0, 0.533);

  text-shadow:
   0 0 10px rgb(201, 240, 201),
   0 0 15px rgb(1, 8, 1),
   0 0 25px rgb(39, 39, 39),
   
   ;
   display: flex;
}

b{

font-size: 1.5rem;
font-family: console;
color: #0f0;
animation: fixText 4s linear infinite;
animation-delay: calc(0.025s * var(--i));




}
@keyframes animatedot {


  0% {
    transform-origin: 100px 50%;
    transform: translate(-50%,-50%) rotate(0deg);
    filter: hue-rotate(0deg);

  }

50% {
    transform-origin: 100px 50%;
    transform: translate(-50%,-50%) rotate(360deg);

  }

  
50.000000000001% {
  transform-origin: -60px 50%;
  transform: translate(-50%,-50%) rotate(360deg);

}


  100% {
    transform-origin: -60px 50%;
    transform: translate(-50%,-50%) rotate(0deg);
    filter: hue-rotate(360deg);

  }
}

@keyframes fixText {


  0% {
    transform: translate(-50%,-50%) rotate(0deg);
    filter: hue-rotate(0deg);

  }

50% {
    transform: translate(-50%,-50%) rotate(360deg);

  }

  
50.000000000001% {
  transform: translate(-50%,-50%) rotate(360deg);

}


  100% {
    transform-origin: -60px 50%;
    transform: translate(-50%,-50%) rotate(0deg);
    filter: hue-rotate(360deg);

  }
}