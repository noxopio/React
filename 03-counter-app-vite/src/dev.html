<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="stylesheet" href="./dev.css">
    <title>Document</title>
</head>
<body>
  <section>
    <div class="box">
      <div class="circle"></div>
      <div class="circle"></div>
      <!-- <div class="dot" style="--i:0;"><b>I</b></div>
      <div class="dot" style="--i:1;"><b>N</b></div>>
      <div class="dot" style="--i:2;"><b>F</b></div>>
      <div class="dot" style="--i:3;"><b>I</b></div>>
      <div class="dot" style="--i:4;"><b>N</b></div>>
      <div class="dot" style="--i:5;"><b>I</b></div>>
      <div class="dot" style="--i:6;"><b>T</b></div>>
      <div class="dot" style="--i:7;"><b>Y</b></div>
      <div class="dot" style="--i:8;"><b>.</b></div>
      <div class="dot" style="--i:9;"><b>.</b></div> -->
      <div class="dot" style="--i:10;"></div>
      <div class="dot" style="--i:11;"></div>
      <div class="dot" style="--i:12;"></div>
      <div class="dot" style="--i:13;"></div>
      <div class="dot" style="--i:14;"></div>
      <div class="dot" style="--i:15;"></div>
      <div class="dot" style="--i:16;"></div>
      <div class="dot" style="--i:17;"></div>
      <div class="dot" style="--i:18;"></div>
      <div class="dot" style="--i:19;"></div>
      <div class="dot" style="--i:20;"></div>
      <div class="dot" style="--i:21;"></div>
      <div class="dot" style="--i:22;"></div>
      <div class="dot" style="--i:23;"></div>
      <div class="dot" style="--i:24;"></div>
      <div class="dot" style="--i:25;"></div>
      <div class="dot" style="--i:26;"></div>
      <div class="dot" style="--i:27;"></div>
      <div class="dot" style="--i:28;"></div>
      <div class="dot" style="--i:29;"></div>
      <div class="dot" style="--i:30;"></div>
      <div class="dot" style="--i:31;"></div>
      <div class="dot" style="--i:32;"></div>
      <div class="dot" style="--i:33;"></div>
      <div class="dot" style="--i:34;"></div>
      <div class="dot" style="--i:35;"></div>
      <div class="dot" style="--i:36;"></div>
      <div class="dot" style="--i:37;"></div>
      <div class="dot" style="--i:38;"></div>


    </div>
  </section>


      
      
      
</body>
</html>