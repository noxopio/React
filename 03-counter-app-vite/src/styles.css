html,
body {
    background-color: black;
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 20.8px;
    padding: 4.375rem;
}

button {
    background-color: white;
    color: black;
    padding: .625rem 1.25rem;
    font-size: 20.8px;
    border: none;
    cursor: pointer;
    margin-top: 1.25rem;
    margin-right: 0.5rem;
}
.button-neomorph {
  background-color: white;
    border-radius: .75rem;
    box-shadow: .5rem .5rem 1rem #1a1a1a, -0.5rem -0.5rem 1rem #3c3c3c;
    border: none;
    color: black;
    padding: .9375rem 1.875rem;
    font-size: 20.8px;
    cursor: pointer;
    outline: none;

    transition: box-shadow 0.3s ease;
  }
  
  .button-neomorph:active {
    box-shadow: inset .5rem .5rem 1rem #1a1a1a, inset -0.5rem -0.5rem 1rem #3c3c3c;
  }
  