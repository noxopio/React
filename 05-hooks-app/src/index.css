body{
    padding: 20px;

}
button{
    padding: 10px;
    margin:0  10px 0  0 ;
    background-color: #f1f1f1;
    border: none;
    border-radius: 5px;
    cursor: pointer;
}