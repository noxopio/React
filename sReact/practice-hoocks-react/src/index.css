body {
  background-color: black;
  color: white;
  padding: 2% 5%;
  /* display: flex;
  justify-content: center;
  align-items: center; */
  display: grid;
  place-items: center;
  margin: 0 auto;
  height: 100vh;
}

button {
  margin: .5rem;
}