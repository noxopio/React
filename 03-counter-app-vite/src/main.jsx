import React from 'react';
import ReactDOM from 'react-dom/client';
import './styles.css';
// import { HelloWorldApp } from './HelloWorldApp';
// import { FirstApp } from './FirstApp';
import { CounterApp } from './CounterApp';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/* <HelloWorldApp /> */}
        {/* <FirstApp title='Neo'/> */}
        <CounterApp  value={100} />
    </React.StrictMode>
)