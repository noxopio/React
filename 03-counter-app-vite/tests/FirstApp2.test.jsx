import { getByTestId, render, screen } from "@testing-library/react"
import { FirstApp } from '../src/FirstApp'

describe('Pruebas en <FirstApp>', () => {

    const title = 'Hola, Soy Goku';
    const subTitle = 'Soy un subtitulo';

    test('Debe de hacer match con el snapshot', () => {
        const { container } = render(<FirstApp title={title} />);
        expect(container).toMatchSnapshot();

    })

    test('Debe de mostrar el mensaje "Hola, Soy Goku"', () => {
        render(<FirstApp title={title} />);
        expect(screen.getAllByText(title)).toBeTruthy();

    })
    test('Debe mostrar el titulo en un H1', () => {
        render(<FirstApp title={title} />);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    })


    test('Debe mostrar el Subtitulo enviado por props ', () => {
        render(<FirstApp
            title={title}
            subTitle={subTitle}
        />);
        expect(screen.getAllByText(subTitle).length).toBe(2);
    })

})
