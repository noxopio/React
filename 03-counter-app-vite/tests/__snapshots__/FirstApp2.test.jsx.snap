// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, Soy Goku
  </h1>
  <h2>
     
    Neo si hay name
  </h2>
  <h2>
    No hay subtitulos
  </h2>
  <h2>
    No hay subtitulos
  </h2>
</div>
`;
